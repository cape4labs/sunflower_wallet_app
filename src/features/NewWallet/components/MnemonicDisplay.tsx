import { View } from 'react-native';
import { MnemonicWord } from '../components/MnemonicWord';

type MnemonicDisplayType = {
  mnemonic: string | null;
  className: string;
};

export function MnemonicDisplay({ mnemonic, className }: MnemonicDisplayType) {
  if (!mnemonic) return null;

  const words = mnemonic.split(' ');
  const half = Math.ceil(words.length / 2);

  const leftColumn = words.slice(0, half);
  const rightColumn = words.slice(half);

  return (
    <View
      className={`flex-row h-auto border-[6px] border-custom_border rounded-xl bg-custom_complement my-5 ${className}`}
    >
      <View className="flex-1 flex-col justify-between m-2">
        {leftColumn.map((word, idx) => (
          <MnemonicWord key={word + idx} idx={idx + 1} word={word} />
        ))}
      </View>

      <View className="flex-1 flex-col justify-between m-2">
        {rightColumn.map((word, idx) => (
          <MnemonicWord key={word + idx + half} idx={idx + 1 + half} word={word} />
        ))}
      </View>
    </View>
  );
}
