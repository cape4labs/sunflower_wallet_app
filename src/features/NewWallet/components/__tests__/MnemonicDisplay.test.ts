import { render } from '@testing-library/react-native';
import { MnemonicDisplay } from '../MnemonicDisplay';

it('MnemonicDisplay snapshot', () => {
  const mnemonic = 'word word word word word word word word word word word word';
  const tree = render(MnemonicDisplay({ mnemonic: mnemonic, className: '' })!).toJSON();
  expect(tree).toMatchSnapshot();
});
