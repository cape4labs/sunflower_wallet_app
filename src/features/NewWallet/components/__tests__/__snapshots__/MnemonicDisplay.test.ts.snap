// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicDisplay snapshot 1`] = `
<View
  className="flex-row h-auto border-[6px] border-custom_border rounded-xl bg-custom_complement my-5 "
>
  <View
    className="flex-1 flex-col justify-between m-2"
  >
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        1
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        2
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        3
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        4
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        5
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        6
        . 
        word
      </Text>
    </View>
  </View>
  <View
    className="flex-1 flex-col justify-between m-2"
  >
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        7
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        8
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        9
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        10
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        11
        . 
        word
      </Text>
    </View>
    <View
      className="flex-row w-full px-2 p-4 my-1 bg-custom_border justify-center rounded-lg"
    >
      <Text
        className="flex-1 p-2 text-white rounded-md text-lg"
      >
        12
        . 
        word
      </Text>
    </View>
  </View>
</View>
`;
